src/hierarchy/CMakeLists.txt

```diff
@@ -25,12 +25,14 @@ add_library(
 add_executable(
     battle
     main.cpp
-)
+  )
 
 target_link_libraries(
     battle
     PRIVATE
         bestiary
-)
+  )
 
-add_subdirectory(tests)
+if(BUILD_TESTING)
```

Comment:
Does it mean I need to use -DBUILD_TESTING in CI explicitly?

Reply:
No. CTest module calls `enable_testing()` by default, so it's `ON` by default. But when it's `OFF` better not to do things related to testing and save some configuring/building time.

Reply:
Are you sure? I've just tested
enable_testing()
```
if(BUILD_TESTING)
  add_subdirectory(tests)
endif()

cmake -DCMAKE_BUILD_TYPE=Debug . 
cmake --build .
```
it didn't create any unittests at all